redis/tests/test_sentinel.rs

```diff
@@ -0,0 +1,404 @@
+mod support;
+
+use redis::sentinel::SentinelClient;
+
+use crate::support::*;
+
+#[test]
+fn test_sentinel_basics() {
+    let cluster = TestSentinelContext::new(3, 1, 3);
+    let sentinel = cluster.sentinel();
+
+    let mut master_con = sentinel
+        .master_for("master1")
+        .unwrap()
+        .get_connection()
+        .unwrap();
+
+    redis::cmd("SET")
+        .arg("{x}key1")
+        .arg(b"foo")
+        .execute(&mut master_con);
+    redis::cmd("SET")
+        .arg(&["{x}key2", "bar"])
+        .execute(&mut master_con);
+
+    assert_eq!(
+        redis::cmd("MGET")
+            .arg(&["{x}key1", "{x}key2"])
+            .query(&mut master_con),
+        Ok(("foo".to_string(), b"bar".to_vec()))
+    );
+}
+
+#[test]
+fn test_sentinel_read_from_random_replica() {
+    let cluster = TestSentinelContext::new(3, 1, 3);
+    let sentinel = cluster.sentinel();
+
+    let mut master_con = sentinel
+        .master_for("master1")
+        .unwrap()
+        .get_connection()
+        .unwrap();
+
+    let mut replica_con = sentinel
+        .replica_for("master1")
+        .unwrap()
+        .get_connection()
+        .unwrap();
+
+    // Write commands would go to the primary nodes
+    redis::cmd("SET")
+        .arg("{x}key1")
+        .arg(b"foo")
+        .execute(&mut master_con);
+    redis::cmd("SET")
+        .arg(&["{x}key2", "bar"])
+        .execute(&mut master_con);
+
+    // Read commands would go to the replica nodes
+    assert_eq!(
```

Comment:
this test will be flakey, since it assumes that the replica will sync in time. Instead of testing set & get (here and in the previous test), just send an `INFO REPLICATION` command and check that the node is indeed a master or replica, and if it is a replica, that it is a replica of the correct master.

Reply:
Ok, makes sense, I'll change it now!